prestoadmin/__init__.py

```diff
@@ -17,30 +17,89 @@
 
 __version__ = '1.2'  # Make sure to update setup.py too
 
+
 import os
+import sys
+
+main_dir = os.path.dirname(os.path.abspath(os.path.dirname(__file__)))
 
 from fabric.api import env
 
+import fabric_patches
+import mode
+
+from prestoadmin.mode import get_mode, for_mode, MODE_STANDALONE, MODE_SLIDER
+from prestoadmin.util.exception import ConfigFileNotFoundError, \
+    ConfigurationError
+
+
+#
+# Subcommands common to all modes. If anybody knows why fabric_patches is in
+# the list, I'll make a note for the next person.
+#
+__all__ = ['fabric_patches', 'mode']
+
+cfg_mode = None
+try:
+    cfg_mode = get_mode()
+except ConfigFileNotFoundError as e:
+    print >>sys.stderr, \
+        "Select a mode using the subcommand 'mode select <mode>'"
+except ConfigurationError as e:
+    print >>sys.stderr, e.message
+
+
+ADDITIONAL_TASK_MODULES = {
+    MODE_SLIDER: [('yarn_slider.server', 'server'),
+                  ('yarn_slider.slider', 'slider')],
+    MODE_STANDALONE: ['topology',
+                      ('configure_cmds', 'configuration'),
+                      'server',
+                      'connector',
+                      'package',
+                      'collect',
+                      'script',
+                      'plugin']}
+
+
+if cfg_mode is not None:
```

Comment:
A better way to handle this might have been in main.py (though you'd have to mess with how Fabric loads tasks), since having this much logic in init.py is gross, but I'm not going to insist that you change it.